src/validation/stmt_validator.rs

```diff
@@ -46,6 +46,51 @@ impl StatementValidator {
             } => {
                 self.validate_cast_literal(target, type_name, location, context);
             }
+            AstStatement::QualifiedReference { elements, .. } => {
+                if elements.len() > 1 {
```

Comment:
this becomes more elegant with an reverse iterator:
no need for len-check and 2 if lets
```
let i = elements.iter().rev();
if let Some(direct_access, accessed_variable) = i.next().zip(i.next()) {

}
```